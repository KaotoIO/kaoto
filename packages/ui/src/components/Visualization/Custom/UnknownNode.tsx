import {
  Alert,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardTitle,
  CodeBlock,
  CodeBlockCode,
} from '@patternfly/react-core';
import { stringify } from 'yaml';
import { FunctionComponent } from 'react';

interface UnknownNodeProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  model: any;
}

export const UnknownNode: FunctionComponent<UnknownNodeProps> = (props) => {
  return (
    <Card>
      <CardHeader>
        <CardTitle>Node source</CardTitle>
      </CardHeader>
      <CardBody>
        <CodeBlock>
          <CodeBlockCode>{stringify(props.model)}</CodeBlockCode>
        </CodeBlock>
      </CardBody>
      <CardFooter>
        <Alert variant="warning" title="Unknow node type">
          The configuration for an unknown node cannot be changed in the configuration form. Please switch to the source
          code and correct the node type. Another option is to replace the step in the graphical editor but in that case
          you will probably lose the existing configuration.
        </Alert>
      </CardFooter>
    </Card>
  );
};
