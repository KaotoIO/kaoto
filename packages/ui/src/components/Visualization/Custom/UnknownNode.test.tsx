import { render } from '@testing-library/react';
import { UnknownNode } from './UnknownNode';

describe('UnknownNode', () => {
  it('renders UnknownNode component', () => {
    const wrapper = render(<UnknownNode model={{}} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('renders UnknownNode component with a model', () => {
    const wrapper = render(<UnknownNode model={{ prop1: 'value 1', prop2: 'value 2' }} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
