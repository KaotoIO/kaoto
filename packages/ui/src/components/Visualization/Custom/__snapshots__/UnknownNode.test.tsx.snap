// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownNode renders UnknownNode component 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-card"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF6/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-v6-c-card__header"
    >
      <div
        class="pf-v6-c-card__header-main"
      >
        <div
          class="pf-v6-c-card__title"
        >
          <div
            class="pf-v6-c-card__title-text"
          >
            Node source
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-c-card__body"
    >
      <div
        class="pf-v6-c-code-block"
      >
        <div
          class="pf-v6-c-code-block__content"
        >
          <pre
            class="pf-v6-c-code-block__pre"
          >
            <code
              class="pf-v6-c-code-block__code"
            >
              {}

            </code>
          </pre>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-c-card__footer"
    >
      <div
        class="pf-v6-c-alert pf-m-warning"
        data-ouia-component-id="OUIA-Generated-Alert-warning-1"
        data-ouia-component-type="PF6/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-v6-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-v6-c-alert__title"
        >
          <span
            class="pf-v6-screen-reader"
          >
            Warning alert:
          </span>
          Unknow node type
        </h4>
        <div
          class="pf-v6-c-alert__description"
        >
          The configuration for an unknown node cannot be changed in the configuration form. Please switch to the source code and correct the node type. Another option is to replace the step in the graphical editor but in that case you will probably lose the existing configuration.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UnknownNode renders UnknownNode component with a model 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-card"
    data-ouia-component-id="OUIA-Generated-Card-2"
    data-ouia-component-type="PF6/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-v6-c-card__header"
    >
      <div
        class="pf-v6-c-card__header-main"
      >
        <div
          class="pf-v6-c-card__title"
        >
          <div
            class="pf-v6-c-card__title-text"
          >
            Node source
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-c-card__body"
    >
      <div
        class="pf-v6-c-code-block"
      >
        <div
          class="pf-v6-c-code-block__content"
        >
          <pre
            class="pf-v6-c-code-block__pre"
          >
            <code
              class="pf-v6-c-code-block__code"
            >
              prop1: value 1
prop2: value 2

            </code>
          </pre>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-c-card__footer"
    >
      <div
        class="pf-v6-c-alert pf-m-warning"
        data-ouia-component-id="OUIA-Generated-Alert-warning-2"
        data-ouia-component-type="PF6/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-v6-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-v6-c-alert__title"
        >
          <span
            class="pf-v6-screen-reader"
          >
            Warning alert:
          </span>
          Unknow node type
        </h4>
        <div
          class="pf-v6-c-alert__description"
        >
          The configuration for an unknown node cannot be changed in the configuration form. Please switch to the source code and correct the node type. Another option is to replace the step in the graphical editor but in that case you will probably lose the existing configuration.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
